sbncode/SBNEventWeight/App/SystToolsEventWeight_module.cc

```diff
@@ -0,0 +1,238 @@
+                      << "[SystToolsEventWeight::produce]       responses.size() of this resp (produced) = " << (*it).responses.size() << std::endl;
+          }
+          if(sph.paramVariations.size()!=(*it).responses.size()){
+            std::cerr << "[SystToolsEventWeight::produce]       prettyName of this resp = " << prettyName << std::endl;
+            std::cerr << "[SystToolsEventWeight::produce]       We expect to have " << sph.paramVariations.size() << " universes, but "
+                      << (*it).responses.size() << " are produced. "
+                      << "Probably this particular event is not relevant to this systematic variation." << std::endl;
+            throw std::exception();
+          }
+
+          std::vector<float> wgts;
+          for( unsigned int i = 0; i < (*it).responses.size(); ++i ){
+
+            if(fDebugMode) std::cout << "[SystToolsEventWeight::produce]         (*it).responses[i] = " << (*it).responses[i] << std::endl;
+            wgts.push_back( (*it).responses[i] );
+
+          } // END loop over universes of this knob
+
+          mcwgh[sp->GetFullyQualifiedName()+"_"+prettyName] = wgts;
+
+        } // END loop over knobs
+
+        mcwghvec->push_back(std::move(mcwgh));
+        art::Ptr<sbn::evwgh::EventWeightMap> wghPtr = makeWeightPtr(mcwghvec->size() - 1);
+        wghassns->addSingle(mclist.at(nMCTruthIndex), wghPtr);
+
+        nMCTruthIndex++;
+
+      } // END loop over MCTruth particles
+
+    } // END systprovider loop
+
+  }
+
+  e.put(std::move(mcwghvec));
+  e.put(std::move(wghassns));
+
+}
+
+
+void SystToolsEventWeight::beginRun(art::Run& run) {
+
+  auto p = std::make_unique<std::vector<sbn::evwgh::EventWeightParameterSet> >();
+  for( auto &sp : fSystProviders ) {
+    MF_LOG_INFO("SystToolsEventWeight") << "sp->GetToolType() = " << sp->GetToolType() << "\n"
+                                  << "sp->GetFullyQualifiedName() = " << sp->GetFullyQualifiedName() << "\n"
+                                  << "sp->GetInstanceName() = " << sp->GetInstanceName() << "\n"
+                                  << "Printing each SystParamHeader of this ISystProviderTool..";
+    //==== Note: typedef std::vector<SystParamHeader> SystMetaData;
+    auto smd = sp->GetSystMetaData();
+    for( auto &sph : smd ){
+      std::cout << "  sph.prettyName = " << sph.prettyName << std::endl;
+
+      std::string rwmode = "multisigma";
+      if(sph.isRandomlyThrown) rwmode = "multisim";
+
+      std::cout << "  rwmode = " << rwmode << std::endl;
+
+      std::vector<float> withds;
+      std::cout << "    paramVariation = ";
+      for(auto pV : sph.paramVariations){
+        std::cout << pV << " ";
+        withds.push_back(pV);
+      }
+      std::cout << std::endl;
+
+      sbn::evwgh::EventWeightParameterSet fParameterSet;
+      fParameterSet.AddParameter(sph.prettyName, withds);
```

Comment:
Moving the data might save a copy. Up to you...
```suggestion
      fParameterSet.AddParameter(sph.prettyName, std::move(withds));
```

Reply:
Updated as suggested.